src/PerfView/PerfViewData.cs

```diff
@@ -1511,6 +1511,238 @@ protected override void WriteHtmlBody(TraceLog dataFile, TextWriter writer, stri
                 AddGenericStopEventToRequest(traceEvent.ContextId, traceEvent);
             };
 
+            var aspNet = new AspNetTraceEventParser(dispatcher);
+
+            // The logic used here is that delays between "AspNetTrace/AspNetReq/Start" and "AspNetTrace/AspNetReq/AppDomainEnter"
+            // will be due to the delay introduced due to the CLR threadpool code based on how
+            // ASP.NET code emits these events.
+            aspNet.AspNetReqStart += delegate (AspNetStartTraceData traceEvent)
+            {
```

Comment:
You can handle all the starts (enter) events with 

aspNet.AddCallbackForEvents(name => name.EndsWith("Enter"), null, delegate(TraceEvent data ) {

       object contextObj = data.PayloadByName("ContextId");
       if (contextObj != null && contextObj is typeof(Guid))
              Guid contextGuid = (Guid) contextObj;
   
               // Code below 
}

You can do something similar for Exit.   that reduces the number of cases dramatically.

Reply:
@vancem awesome !!! Will try this out